### PR TITLE
fix: csv export dashboard filters

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: #7068 

### Description:

OG chart:
CleanShot 2023-09-26 at 20 23 35@2x

dashboard with chart + dashboard filter:
CleanShot 2023-09-26 at 20 23 17@2x

old buggy csv:
CleanShot 2023-09-26 at 20 22 59@2x

new fixed csv:
CleanShot 2023-09-26 at 20 22 51@2x

### CODE REVIEW

File: packages/common/src/utils/filters.ts [hunk 1]
```diff
@@ -376,3 +379,103 @@ export const getFilterRulesByFieldType = (
...
+        dashboardFiltersForTile && dashboardFiltersForTile.metrics.length > 0
+            ? {
+                  id: 'no',
+                  and: [
+                      ...(metricQuery.filters.metrics
+                          ? [metricQuery.filters.metrics]
+                          : []),
+                      ...dashboardFiltersForTile.metrics,
+                  ],
+              }
```
Comment (IrakliJani) [inline packages/common/src/utils/filters.ts:469]: hey @ZeRego is this the right way to combine filters? in this case dashboard filters to the existing metric filters? if yes, does it matter what the id's will be? I copy pasted this from here: https://github.com/jayanand05/lightdash/blob/360ebbcaf1b74b49ff982193d036f1b4e149bff0/packages/frontend/src/hooks/dashboard/useSavedQueryWithDashboardFilters.ts#L62

I was also thinking to extract this as a function, and name it something like, "combineFilters"

Comment (ZeRego) [reply packages/common/src/utils/filters.ts:469]: > is this the right way to combine filters?

Yes that is correct. 
We also have a static method `ProjectService.combineFilters` that does almost the same.

> does it matter what the id's will be?

not at the moment

**Review by magnew — APPROVED** [review]
This looks good to me. I tested the FE dashboard filter side of this and everything works right.

File: packages/common/src/utils/filters.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/common/src/utils/filters.ts:469]: @ZeRego great, as this isn't going to hit the DB and the feature's working - I'm gonna merge it. [Created a ticket](https://github.com/lightdash/lightdash/issues/7246) to tackle duplicated code tomorrow. which will combine `combineFilters` function for both frontend and backend 👌

**Pull request merged**